python/ray/monitor.py

```diff
@@ -361,26 +360,22 @@ def run(self):
     level = logging.getLevelName(args.logging_level.upper())
     logging.basicConfig(level=level, format=args.logging_format)
 
-    redis_ip_address = get_ip_address(args.redis_address)
-    redis_port = get_port(args.redis_address)
-
     if args.autoscaling_config:
         autoscaling_config = os.path.expanduser(args.autoscaling_config)
     else:
         autoscaling_config = None
 
     monitor = Monitor(
-        redis_ip_address,
-        redis_port,
+        args.redis_address,
         autoscaling_config,
         redis_password=args.redis_password)
 
     try:
         monitor.run()
     except Exception as e:
         # Something went wrong, so push an error to all drivers.
-        redis_client = redis.StrictRedis(
-            host=redis_ip_address, port=redis_port)
+        redis_client = ray.runner.create_redis_client(
+            args.redis_address, password=args.redis_password)
```

Comment:
Here is a bug fixing: The original code forget to use Redis password when processing errors.